### DISCUSSION TITLE
Problem with practical 4

### DISCUSSION THREAD

**Discussion by EleonoraMilea** [IBAHCM/RPiR#491] [Archive] [answered]
Hi @richardreeve and @soniamitchell,
I am having issues with my function for practical 4.
The function seem to work however it gives me the wrong results. These are the results I get when I use qs= c(0:10) on tree.pop
2.990000e+02   5.565416e+01   2.225266e+01  4.987118e+04    6.602126e+09    5.006857e+16    2.178790e+25    5.435949e+35  7.737891e+47   6.240402e+61

You can see that the numbers are technically correct because for instance, species richness is 299. However, they are very large numbers and I am not sure why. 
The general function for practical 3 that I am calling in my practical 4 function is working properly as it gives me the correct results, so I am not sure what might be wrong here. 

Any ideas? 

Thank you

**Comment by richardreeve** [comment, marked as answer]
@EleonoraMilea - does the function in practical 3 definitely give the right answers for `q = 3`, `q = 4`, etc? The diversity should be dropping for `tree.pop` as q increases if you're uncertain. If not, the problem is obviously there...

If that function is definitely working, then it seems like the explanation must be a problem with your loop (if you have one?) in practical 4. If you really can't solve this, don't worry too much - if your functions work and are documented and the package works, but in some cases the results are wrong, you'll only get penalised once for the mistake.

**Reply by EleonoraMilea** [reply]
Hi Richard, thank you, this helped! My function 3 was indeed wrong. I had an issue with the (1/(1-q)) parentheses. 

Thanks again :)
